### PR TITLE
Move await keyword to inside of function

### PR DESCRIPTION
**Pull request by Cow258**
Fixes [#28811](https://github.com/Microsoft/TypeScript/issues/28811)

### CODE REVIEW

**Comment by Cow258** [comment]
image

**Comment by Cow258** [comment]
@weswigham

File: src/services/completions.ts [hunk 1]
```diff
@@ -1957,7 +1957,7 @@ namespace ts.Completions {
     }
 
     function isFunctionLikeBodyKeyword(kind: SyntaxKind) {
-        return kind === SyntaxKind.AsyncKeyword || !isContextualKeyword(kind) && !isClassMemberCompletionKeyword(kind);
+        return kind === SyntaxKind.AsyncKeyword || kind === SyntaxKind.AwaitKeyword || !isContextualKeyword(kind) && !isClassMemberCompletionKeyword(kind);
```
Comment (sheetalkamat) [inline src/services/completions.ts]: I don't think this is right fix. You want to create `AsyncFunctionLikeBodyKeywords` and add await in there to ensure that await is offered only in `async` functions? Also remove it from global completions?

Comment (sheetalkamat) [reply src/services/completions.ts]: Also please add a test case.

Comment (Cow258) [reply src/services/completions.ts]: I dont think i have enough skill to remove await from global keywords and create AsyncFunctionLikeBodyKeywords.
I had no idea about how to adding AsyncFunctionLikeBodyKeywords

Comment (Cow258) [reply src/services/completions.ts]: BTW, For now that maybe a shortcut to adding `await` keyword to inside of function. I am working so smooth with that fix by local build.

Comment (Cow258) [reply src/services/completions.ts]: I think that...

- Typescript 3.2.2 (Quick Fix)

  1. Add `await` inside of function for quick fix
  2. At least people can use it as quickly as possible

Then

- Typescript 3.3.0 (Deep Fix)

  1. Create AsyncFunctionLikeBodyKeywords
  2. Remove `await` from `globalKeywords`
  3. Remove `await` from `normal function`
  4. Add `await` inside of `async function` only

Comment (sheetalkamat) [reply src/services/completions.ts]: @DanielRosenwasser To triage on if  we want this fix as suggested by @Cow258 for 3.2.2?

Comment (Cow258) [reply src/services/completions.ts]: > Also please add a test case.

After this fix `await` keyword will be offered global and inside of function.
await

Comment (DanielRosenwasser) [reply src/services/completions.ts]: It sounds like you already have the right direction for the fix, so let's just do that today and get it into 3.2.2.

**Pull request merged**